controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -369,3 +377,70 @@ func (p *OpenAPIProductReconciler) desiredPrivateAPISecurity() *capabilitiesv1be
 
 	return privateAPISec
 }
+
+func (p *OpenAPIProductReconciler) desiredOIDCSecurity() *capabilitiesv1beta1.SecuritySpec {
+	if p.openapiCR.Spec.SecurityScheme == nil && p.openapiCR.Spec.SecurityScheme.Type == nil {
+		return nil
+	}
+
+	privateAPISec := &capabilitiesv1beta1.SecuritySpec{}
+
+	if *p.openapiCR.Spec.SecurityScheme.Type == "openIdConnect" &&
+		p.openapiCR.Spec.SecurityScheme.OpenIdConnectUrl != nil {
+		privateAPISec.OpenApiSecuritySchemeType = p.openapiCR.Spec.SecurityScheme.Type
+		privateAPISec.OpenIdConnectUrl = p.openapiCR.Spec.SecurityScheme.OpenIdConnectUrl
+	}
+
+	return privateAPISec
+}
+
+func (p *OpenAPIProductReconciler) desiredOIDCAuthentication(secReq *helper.ExtendedSecurityRequirement) *capabilitiesv1beta1.AuthenticationSpec {
+	authSpec := &capabilitiesv1beta1.AuthenticationSpec{
+		OIDC: &capabilitiesv1beta1.OIDCSpec{
+			IssuerType: "keycloak",
```

Comment:
This should be configurable from the OpenAPI CR

Reply:
updated. Thank you